src/root.zig

```diff
@@ -210,6 +210,15 @@ pub fn AlgorithmType(
             self.max_needle = max_needle;
         }
 
+        // Check is there is enough memory allocated
+        pub fn hasSize(self: *Self, max_haystack: usize, max_needle: usize) bool {
```

Comment:
```suggestion
        pub fn hasSize(self: *const Self, max_haystack: usize, max_needle: usize) bool {
```
This won't and should not mutate memory, so we mark the pointer as `const`.